pybop/optimisers/base_optimiser.py

```diff
@@ -3,78 +3,109 @@ class BaseOptimiser:
     A base class for defining optimisation methods.
 
     This class serves as a template for creating optimisers. It provides a basic structure for
-    an optimisation algorithm, including the initial setup and a method stub for performing
-    the optimisation process. Child classes should override the optimise and _runoptimise
-    methods with specific algorithms.
+    an optimisation algorithm, including the initial setup and a method stub for performing the
+    optimisation process. Child classes should override update_options and the_run method with
+    a specific algorithm.
     """
 
-    def __init__(self, bounds=None):
+    def __init__(self, x0=None, sigma0=None, bounds=None):
         """
-        Initializes the BaseOptimiser.
+        Initialises the BaseOptimiser.
 
         Parameters
         ----------
+        x0 : array_like
+            Initial position from which optimisation will start.
+        sigma0 : float, optional
+            Initial step size or standard deviation depending on the optimiser.
         bounds : sequence or Bounds, optional
-            Bounds on the parameters. Default is None.
+            Bounds on the parameters (default: None).
         """
+        self._cost_function = None
+        self.x0 = x0
+        self.sigma0 = sigma0
         self.bounds = bounds
+
+        self.log = []
+        self._max_iterations = None
+        self.set_max_iterations()
+
+    def update_options(self, **optimiser_kwargs):
```

Comment:
To align with standard set/get interfacing. Possibly worth adding a get_options as well.
```suggestion
    def set_options(self, **optimiser_kwargs):
```